libs/langchain_v1/langchain/agents/factory.py

```diff
@@ -158,9 +159,7 @@ def inner_handler(req: ModelRequest) -> ModelResponse:
         return composed
 
     # Compose right-to-left: outer(inner(innermost(handler)))
-    result = handlers[-1]
-    for handler in reversed(handlers[:-1]):
-        result = compose_two(handler, result)
+    result = reduce(lambda handler1, handler2: compose_two(handler2, handler1), reversed(handlers))
```

Comment:
@arun-geek we typically have ~1-10 handlers... have you tried benchmarking to confirm that:

1. The change has the effect you expect?
2. That it's actually a bottleneck? i.e., any improvements here may have no practical effect.